File: app/src/main/java/com/estasvegano/android/estasvegano/AppComponent.java
package com.estasvegano.android.estasvegano;

import android.support.annotation.NonNull;

import com.estasvegano.android.estasvegano.data.DataModule;
import com.estasvegano.android.estasvegano.model.ModelModule;
import com.estasvegano.android.estasvegano.view.AddProductFragment;
import com.estasvegano.android.estasvegano.view.CodeReaderFragment;
import com.estasvegano.android.estasvegano.view.ViewProductFragment;
import com.estasvegano.android.estasvegano.view.dialog.ComplainProductDialogFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {DataModule.class, AppModule.class, ModelModule.class})
public interface AppComponent {

    void inject(@NonNull CodeReaderFragment codeReaderFragment);

    void inject(@NonNull ViewProductFragment viewProductFragment);

    void inject(@NonNull ComplainProductDialogFragment complainProductDialogFragment);

    void inject(@NonNull AddProductFragment addProductFragment);
}

File: app/src/main/java/com/estasvegano/android/estasvegano/AppComponent.kt
package com.estasvegano.android.estasvegano

import com.estasvegano.android.estasvegano.data.DataModule
import com.estasvegano.android.estasvegano.model.ModelModule
import com.estasvegano.android.estasvegano.ui.AddProductFragment
import com.estasvegano.android.estasvegano.ui.CodeReaderFragment.CodeReaderComponent
import com.estasvegano.android.estasvegano.ui.CodeReaderFragment.CodeReaderModule
import com.estasvegano.android.estasvegano.ui.ViewProductFragment
import com.estasvegano.android.estasvegano.ui.dialog.ComplainProductDialogFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(DataModule::class, AppModule::class, ModelModule::class))
interface AppComponent {

    fun inject(viewProductFragment: ViewProductFragment)

    fun inject(complainProductDialogFragment: ComplainProductDialogFragment)

    fun inject(addProductFragment: AddProductFragment)

    fun plus(codeReaderModule: CodeReaderModule): CodeReaderComponent
}
